Packs/AggregatedScripts/Scripts/ExpirePassword/ExpirePassword.py

```diff
@@ -0,0 +1,544 @@
+            - list[dict]: The raw command results from ad-get-user.
+            - str: The human-readable output from the ad-get-user command.
+    """
+    args_get_user = {"username": username, "using": using}
+    res_get_user, hr_get_user = run_command("ad-get-user", args_get_user)
+
+    dont_expire_flag = True
+    for res in res_get_user:
+        entry_context = res.get("EntryContext", {})
+
+        # Get the ActiveDirectory.Users key (may have filter syntax like "ActiveDirectory.Users(obj.dn == val.dn)")
+        output_key = None
+        for key in entry_context:
+            if key.startswith("ActiveDirectory.Users"):
+                output_key = key
+                break
+
+        if output_key:
+            outputs = entry_context.get(output_key, [])
+            # If outputs is a list, get the first element
+            if isinstance(outputs, list) and outputs:
+                user_data = outputs[0]
+            elif isinstance(outputs, dict):
+                user_data = outputs
+            else:
+                continue
+
+            # Extract userAccountControlFields
+            fields = user_data.get("userAccountControlFields", {})
+            dont_expire_flag = fields.get("DONT_EXPIRE_PASSWORD", None)
+            break
+
+    return dont_expire_flag, res_get_user, hr_get_user
+
+
+def run_active_directory_query_v2(user: UserData, using: str) -> tuple[list[ExpiredPasswordResult], str]:
+    """
+    Expires a user's password in Active Directory by first checking if the 'Password Never Expires' flag
+    is set, and only proceeding with password expiration if the flag is false.
+
+    Args:
+        user (UserData): The user data dictionary.
+        using (str): The name of the Active Directory integration instance.
+
+    Returns:
+        tuple[list[ExpiredPasswordResult], str]: A list containing the result of the password expiration operation and the HR.
+    """
+    username = user["Username"]
+
+    # 1. Check if the "Password Never Expires" flag is set
+    dont_expire_flag, res_get_user, hr_get_user = check_ad_password_never_expires(username, using)
+
+    # 2. If the flag is True, return failure with informative message
+    if dont_expire_flag is True:
+        return [
+            ExpiredPasswordResult(
+                Result="Failed",
+                Message=(
+                    f"Cannot expire password for user {username} due to user policy. "
+                    f"The 'Password Never Expire' flag is set to true. "
+                    f"To expire the password, please change this setting to false using the command "
+                    f"'ad-modify-password-never-expire'."
+                ),
+                Instance=get_instance_from_result(res_get_user[0]) if res_get_user else "",
+            )
+        ], hr_get_user
+
+    # 3. If the flag is None, return failure (unable to retrieve the flag)
+    if dont_expire_flag is None:
```

Comment:
I'm not sure about this use case. Is it valid? 
Why will it return None? 
Is there any case that this field won't appear in the context?